File: test-whisper.py
#!/usr/bin/env python3
import whisper
import sys
import time

def test_whisper(audio_file=None):
    print("🎙️ Testing Whisper transcription...")
    
    # Load model
    print("Loading model...")
    start_time = time.time()
    model = whisper.load_model("base", device="mps")
    load_time = time.time() - start_time
    print(f"✓ Model loaded in {load_time:.2f} seconds")
    
    if audio_file:
        print(f"\nTranscribing: {audio_file}")
        start_time = time.time()
        result = model.transcribe(audio_file)
        transcribe_time = time.time() - start_time
        
        print(f"\n✓ Transcription completed in {transcribe_time:.2f} seconds")
        print(f"Language: {result['language']}")
        print(f"Text preview: {result['text'][:200]}...")
    else:
        print("\n✓ Whisper is ready to use!")
        print("To test with an audio file: python test-whisper.py <audio_file>")

if __name__ == "__main__":
    audio_file = sys.argv[1] if len(sys.argv) > 1 else None
    test_whisper(audio_file)

File: scripts/test-whisper.ts
#!/usr/bin/env npx tsx

import { whisperService } from '../src/services/transcription/whisper.service';
import { logger } from '../src/utils/logger';

async function testWhisper() {
  console.log('Testing Whisper service initialization...\n');
  
  try {
    // Test if Whisper is available
    console.log('1. Checking if Whisper service is available...');
    const isAvailable = await whisperService.isAvailable();
    console.log(`   ✅ Whisper service available: ${isAvailable}\n`);
    
    if (!isAvailable) {
      console.error('   ❌ Whisper service is not available. Please install dependencies.\n');
      return;
    }
    
    // Get supported formats
    console.log('2. Checking supported audio formats...');
    const formats = whisperService.getSupportedFormats();
    console.log(`   ✅ Supported formats: ${formats.join(', ')}\n`);
    
    // Test with a small sample (if provided)
    const testAudioUrl = process.argv[2];
    if (testAudioUrl) {
      console.log('3. Testing transcription with provided audio URL...');
      console.log(`   Audio URL: ${testAudioUrl}`);
      console.log('   This may take a few minutes...\n');
      
      const startTime = Date.now();
      const result = await whisperService.transcribeAudio(testAudioUrl);
      const duration = (Date.now() - startTime) / 1000;
      
      console.log('   ✅ Transcription completed!');
      console.log(`   - Processing time: ${duration.toFixed(2)} seconds`);
      console.log(`   - Text length: ${result.text.length} characters`);
      console.log(`   - Language: ${result.language || 'auto-detected'}`);
      console.log(`   - First 200 chars: ${result.text.substring(0, 200)}...`);
    } else {
      console.log('\n💡 Tip: You can test transcription by providing an audio URL:');
      console.log('   npm run test:whisper <audio-url>');
    }
    
    console.log('\n✅ All tests passed! Whisper is properly configured.');
    
  } catch (error) {
    console.error('\n❌ Whisper test failed:', error);
    console.error('\nTroubleshooting steps:');
    console.error('1. Ensure Python 3 is installed: python3 --version');
    console.error('2. Install Whisper: pip3 install openai-whisper');
    console.error('3. For M1 Macs, install MLX: pip3 install mlx-whisper');
    console.error('4. Check the WHISPER_* environment variables in .env');
  }
}

// Run the test
testWhisper().catch(console.error);